### DISCUSSION TITLE
Multiple nuxt.config files and how to handle it?

### DISCUSSION THREAD

**Discussion by ionutciochinaru** [nuxt/nuxt#28103] [Questions] [answered]
I have an app where i need to setup configs for different country's with different variables.
My question is how do i handle it so in runtimeConfig i have specific variables for specific run modes.
I want to be able to npm run dev:countryName1 or npm run build:countryName1 and use the variables from that specific file nuxt.config.countryName1.ts for example.

This is what I tried so far with no success.

in my nuxt.config.ts file i made
```
const commonConfig: NuxtConfig = {
  // Common configurations here
  }
export default commonConfig;
```
  
  And in my nuxt.config.countryName1.ts file i destructure commonConfig
  
  ```
export default defineNuxtConfig({
  ...commonConfig,
  runtimeConfig: {
    ...commonConfig.runtimeConfig,
    public: {
      ...commonConfig.runtimeConfig?.public,
      // countryName1 configurations here
```
      then in my package.json
`      "dev:countryName1": "nuxt dev --config-file nuxt.config.countryName1.ts",`
      
 Is there a better approach to this problem?
 
 Thank you!

**Comment by santiagodeheza** [comment, marked as answer]
Very good material  
[Nuxt's runtimeConfig - The most common mistake](https://www.youtube.com/watch?v=_FYV5WfiWvs)
[Understanding runtimeConfig](https://masteringnuxt.com/blog/configuration-in-nuxt-3-runtimeConfig-vs-appConfig)

One approach that comes to my mind is to create env variable files for each country in the project root.

```
.env.countryName1
.env.countryName2
.env.countryName3
```
### .env.countryName1
NUXT_PUBLIC_COUNTRY_CODE=countryName1
NUXT_PUBLIC_API_URL=https://api.countryName1.example.com

### nuxt.config.ts
```js
export default defineNuxtConfig({
  runtimeConfig: {
    public: {
      apiUrl: process.env.API_URL,
    },
  },

});
```

### package.json
```json
{
  "scripts": {
    "dev:countryName1": "nuxt dev --dotenv .env.countryName1",
    "build:countryName1": "nuxt build --dotenv .env.countryName1",
    "dev:countryName2": "nuxt dev --dotenv .env.countryName2",
    "build:countryName2": "nuxt build --dotenv .env.countryName2"
  }
}
```
```sh
npm run dev:countryName1 or npm run build:countryName1
```

There might be other ways, but this one is a straightforward one 😀
